### PR TITLE
Delete account from billing batch

### PR DESCRIPTION
**Pull request by Stephan-Hough**
Feat/WATER-2448
-- feature to remove an account from a batch
-- 2 new routes get and post to delete account with controller methods
-- bacthCreate service merged with batch service to clean up code
-- 2 new methods added to batch service getInvoice and deleteAccount
-- link added from bill summary view to remove account
-- delete account view created

### CODE REVIEW

File: src/internal/modules/billing/controller.js [hunk 1]
```diff
@@ -299,6 +299,31 @@ const getTransactionsCSV = async (request, h) => {
...
+    batch: { id: request.params.batchId },
+    back: `/billing/batch/${batchId}/summary`
+  });
+};
+
+const postBillingBatchDeleteAccount = async (request, h) => {
+  const accountId = request.params.invoiceId;
+  const batchId = request.params.batchId;
+  await services.water.billingBatchService.deleteAccountFromBatch(batchId, accountId);
+  return h.redirect(`/billing/batch/${request.params.batchId}/summary`);
```
Comment (DGajewska) [inline src/internal/modules/billing/controller.js]: It looks like you missed this function when you were writing your tests

File: src/internal/modules/billing/controller.js [hunk 2]
```diff
@@ -299,6 +299,31 @@ const getTransactionsCSV = async (request, h) => {
...
 };
 
+/**
+ * Remove an invoice from the bill run
+ * @param {*} request
+ * @param {*} h
+ */
+const getBillingBatchDeleteAccount = async (request, h) => {
+  const batchId = request.params.batchId;
+  const invoiceId = request.params.invoiceId;
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/modules/billing/controller.js]: done

File: src/internal/modules/billing/controller.js [hunk 3]
```diff
@@ -299,6 +299,31 @@ const getTransactionsCSV = async (request, h) => {
...
+  const account = await batchService.getBatchInvoice(batchId, invoiceId);
+  return h.view('nunjucks/billing/batch-delete-account', {
+    ...request.view,
+    pageTitle: 'You are about to remove this invoice from the bill run',
+    account,
+    batch: { id: request.params.batchId },
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/modules/billing/controller.js]: done

File: src/internal/modules/billing/controller.js (hunk 1 above)
Comment (Stephan-Hough) [reply src/internal/modules/billing/controller.js]: done and i added some unit tests for the other post methods that also didn't have any tests

File: src/internal/views/nunjucks/billing/batch-delete-account.njk [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          <div>{{ licence.licenceRef }}</div>
+        {% endfor %}
+        <div>£{{ account.amount | number}}</div>
+      </div>
+      <form method="POST" action="/billing/batch/{{batch.id}}/removeAccount/{{account.id}}">
+        <input type="hidden" name="csrf_token" value="{{ csrfToken }}" />
+        {{ govukButton({
+          text: "Remove invoice"
+        }) }}
+      </form>
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/views/nunjucks/billing/batch-delete-account.njk]: done

**Pull request merged**